lib/puppet/provider/acl/windows.rb

```diff
@@ -113,8 +113,11 @@ def permissions_to_s(permissions)
     perms = permissions.select { |p| !p.is_inherited}
 
     unless perms.nil?
-      perms.each do |perm|
-        perm.identity = get_account_name(perm.identity)
+      perms = perms.map do |perm|
+        perm_hash = perm.to_hash
+        perm_hash['identity'] = get_account_name(perm.identity)
+        # TODO: is it just as easy to perm.dup instead?
+        Puppet::Type::Acl::Ace.new(perm_hash, nil)
```

Comment:
Look into just returning an array of strings / hashes here instead... or calling `.to_s` directly on the Ace object and supplanting the identity.

Note that the behavior on Puppet 4 / Puppet 5 will be different.